internal/logic/message_match/message.go

```diff
@@ -54,13 +56,21 @@ func (t *LogicMessageMatch) GetLatestBlockNumber(ctx context.Context, layer type
 }
 
 // InsertOrUpdateMessageMatches insert or update the gateway/messenger event info
-func (t *LogicMessageMatch) InsertOrUpdateMessageMatches(ctx context.Context, layer types.LayerType, messengerMessageMatches []orm.MessageMatch) error {
+func (t *LogicMessageMatch) InsertOrUpdateMessageMatches(ctx context.Context, layer types.LayerType, gatewayMessageMatches []orm.GatewayMessageMatch, messengerMessageMatches []orm.MessengerMessageMatch) error {
 	var effectRows int64
 	err := t.db.Transaction(func(tx *gorm.DB) error {
```

Comment:
I think there's no need for a transaction here because the block statuses are updated after all the goroutines finish successfully.